tests/components/matter/test_update.py

```diff
@@ -169,3 +194,185 @@ async def test_update_install(
     state = hass.states.get("update.mock_dimmable_light")
     assert state.state == STATE_OFF
     assert state.attributes.get("installed_version") == "v2.0"
+
+
+async def test_update_install_failure(
+    hass: HomeAssistant,
+    matter_client: MagicMock,
+    check_node_update: AsyncMock,
+    update_node: AsyncMock,
+    updateable_node: MatterNode,
+) -> None:
+    """Test update entity service call errors."""
+    state = hass.states.get("update.mock_dimmable_light")
+    assert state
+    assert state.state == STATE_OFF
+    assert state.attributes.get("installed_version") == "v1.0"
+
+    await async_setup_component(hass, "homeassistant", {})
+
+    check_node_update.return_value = MatterSoftwareVersion(
+        vid=65521,
+        pid=32768,
+        software_version=2,
+        software_version_string="v2.0",
+        firmware_information="",
+        min_applicable_software_version=0,
+        max_applicable_software_version=1,
+        release_notes_url="http://home-assistant.io/non-existing-product",
+        update_source=UpdateSource.LOCAL,
+    )
+
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
+        {
+            ATTR_ENTITY_ID: "update.mock_dimmable_light",
+        },
+        blocking=True,
+    )
```

Comment:
How does this normally get triggered? I'd rather see it naturally be updated using either getting a push function from a mock and calling it or skipping time

Reply:
It is polled normally, every 12h:

https://github.com/home-assistant/core/blob/98a007cb2f04eaba6bc28651d964816f45e57724/homeassistant/components/matter/update.py#L73-L76

Using the `update_entity` action is a perfectly valid way to force an update IMHO. But if there is a easy way to test that poll, that is probably closer to the typical use case. :thinking:

Reply:
We can use the freezer to skip 12 hours in time and then it will automatically update

Reply:
Sounds good! I've added a separate test to test the service call still, and use the freezer by default.